### PR TITLE
refactor: first batch atom to ts

### PR DESCRIPTION
**Pull request by Kyl23**
# Important
- GlobalLayout & SearchBar haven't pass eslint

# pending refactor
- PreviewInput
- Select
- TextArea
- ToggleButtonGroup

### CODE REVIEW

File: src/components/atom/propsType.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { KeyboardEventHandler, MouseEventHandler } from 'react';
+
```
Comment (JiaAnTW) [inline src/components/atom/propsType.ts:2]: use  `.d.ts` instead of `.ts`

File: src/components/atom/propsType.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { KeyboardEventHandler, MouseEventHandler } from 'react';
```
Comment (JiaAnTW) [inline src/components/atom/propsType.ts]: use `.d.ts` instead of `.ts`

Comment (Kyl23) [reply src/components/atom/propsType.ts]: noted

File: src/components/atom/Button/index.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+const theme: ThemeType = {
+    primary: {
+        color: 'white',
+        backgroundColor: 'rgb(229, 68, 109)',
+    },
+    light: {
+        color: 'black',
+        backgroundColor: 'lightgray',
+    },
+};
```
Comment (JiaAnTW) [inline src/components/atom/Button/index.ts]: @Kyl23 這邊顏色可以整合進這裡嗎 方便之後做dark mode
https://github.com/ncku-study/ncku-study/blob/65701197d5727c51954d0a06bfc919f35e3afbaa/src/theme/global.js#L4-L16

File: src/components/atom/InputLabel/index.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import style from '@emotion/styled';
+import InputLabel from '@mui/material/InputLabel';
+import { color } from '~/src/theme/global';
```
Comment (Kyl23) [inline src/components/atom/InputLabel/index.ts:3]: 這種要不要直接用props把color傳進來，用一個hook去抓dark跟light的theme

Comment (JiaAnTW) [reply src/components/atom/InputLabel/index.ts:3]: props應該頂多傳theme就好 讓color鎖在style裡面，這樣同theme的元件顏色就會統一 也比較好控制dark/light mode

**Comment by JiaAnTW** [comment]
之後記得補一下test 可以參考 https://github.com/ncku-study/ncku-study/pull/19 感謝~

**Review by JiaAnTW — APPROVED** [review]

**Pull request merged**